x-pack/agent/pkg/agent/application/fleet_gateway.go

```diff
@@ -20,6 +21,11 @@ type agentInfo interface {
 	AgentID() string
 }
 
+type fleetReporter interface {
+	repo.Backend
```

Comment:
From what I see you don't need to have the complete interface imported here, because you are only usings `Events()`?